osquery/tables/system/darwin/extended_attributes.cpp

```diff
@@ -0,0 +1,223 @@
+                         {"value", values[1]}});
+    } else {
+      // Changed to blank because it might not be no data, it might be we just
+      // don't know how to parse because it's corrupted. So I have elected
+      // to not make the No Data decision for the investigator.
+      results.push_back({{"path", path},
+                         {"directory", directory},
+                         {"key", "download_url"},
+                         {"value", ""}});
+      results.push_back({{"path", path},
+                         {"directory", directory},
+                         {"key", "download_page"},
+                         {"value", ""}});
+    }
+  }
+}
+
+// Parse a given XAttrAttribute struct as a Apple Quarantine string
+void parseQuarantineFile(QueryData &results,
+                         const XAttrAttribute &x_att,
+                         const std::string &path,
+                         const std::string &directory) {
+  std::vector<std::string> values = osquery::split(x_att.attribute_data, ";");
+  if (values.size() < 2) {
+    return;
+  }
+  results.push_back({{"path", path},
+                     {"directory", directory},
+                     {"key", "quarantine_creator"},
+                     {"value", values[2]}});
+  results.push_back({{"path", path},
+                     {"directory", directory},
+                     {"key", "quarantine_state"},
+                     {"value", values[0]}});
+}
+
+// Process a file and extract all attribute information, parsed or not.
+void getFileData(QueryData &results,
+                 const std::string &path,
+                 const std::string &directory) {
+  std::vector<std::string> attributes = parseExtendedAttributeList(path);
+  for (const auto &attribute : attributes) {
+    struct XAttrAttribute x_att = getAttribute(path, attribute);
+    try {
+      (*xParseMap.at(attribute))(results, x_att, path, directory);
+    } catch (const std::out_of_range &oor) {
+      // We don't have a function in our map for this key so throw it in
+      // verbatim
+      results.push_back({{"path", path},
+                         {"directory", directory},
+                         {"key", attribute},
+                         {"value", x_att.attribute_data}});
+    }
+  }
+}
+
+QueryData genXattr(QueryContext &context) {
+  QueryData results;
+  auto paths = context.constraints["path"].getAll(EQUALS);
+
+  for (const auto &path_string : paths) {
+    boost::filesystem::path path = path_string;
+    // Folders can have extended attributes too
+    if (!(boost::filesystem::is_regular_file(path) ||
+          boost::filesystem::is_directory(path))) {
+      continue;
+    }
+    getFileData(results, path.string(), path.parent_path().string());
+  }
+
+  auto directories = context.constraints["directory"].getAll(EQUALS);
+  for (const auto &directory_string : directories) {
+    boost::filesystem::path directory = directory_string;
```

Comment:
I don't think you need the `boost::filesystem::path` structure, a `std::string` should be ok?